### ISSUE TITLE
Does Svol in function `volumes` in `DBDD_predict_diag.sage` means the log(norm) of the target vector? 

### ISSUE THREAD

**Issue by Summwer** [lducas/leaky-LWE-Estimator#12]
Hello, developer, 

​     Does Svol in function `volumes` in `DBDD_predict_diag.sage` means the log(norm) of the target vector? If is, then I think maybe there's a mistake in computing the **log(norm) of the target vector**. This mistake might occur while secret vector and noise vector follows different distribution.

​     I see in the file `instance_gen.sage`, you initialize the array `S` in function `initialize_from_LWE_instance`  by `S = diagonal_matrix(m * [s_e] + n * [s_s] + [0])`. 

​     And in the function `integrate_q_vectors`, you decrease the dimension of lwe instance by deleting the q-vector from  large index to small by the function `integrate_short_vector_hint`. If the q-vector could be deleted, then let `S[i] = 0`. However, I test the Kyber scheme($\eta_1 = 3$, $\eta_2$ = 2, m=512, n = 512, intitial_d = 1025), where `s_e=1.` and  `s_s=1.5` in function `initialize_from_LWE_instance`. In function  `integrate_q_vectors` and I print the index `i` in function `integrate_q_vectors`: 

<img width="586" alt="image" src="https://github.com/lducas/leaky-LWE-Estimator/assets/26530870/2c757a96-a26f-4fec-8459-83307a8dadc7">

it outputs i-index accounted from large($d-2$) to small:

<img width="530" alt="image" src="https://github.com/lducas/leaky-LWE-Estimator/assets/26530870/1992b114-9e09-45b3-a728-138fe2db6191">

In function `integrate_short_vector_hint`, if we print i and S[i]:

<img width="304" alt="image" src="https://github.com/lducas/leaky-LWE-Estimator/assets/26530870/aa8f0d2e-ab72-4185-9c46-9413a94f9533">

We can see that  it makes S[i] = 0, where `S[i]=s_s`: 

<img width="561" alt="image" src="https://github.com/lducas/leaky-LWE-Estimator/assets/26530870/90013ba4-0e83-403c-9117-572cd497300c">

Since in LWE instance, we can decrease the samples in LWE to find a small matrix for reduction. Let the original target norm be $a$, I think the target norm should become $a - s\_e$ by deleting a q-vector and one sample in matrix A, but you delete an `s_s` in S, where S is used to compute the `Svol` in function `volumes` in `DBDD_predict_diag.sage`:

<img width="475" alt="image" src="https://github.com/lducas/leaky-LWE-Estimator/assets/26530870/e0dad904-c8b7-4bd1-9a96-6d84f50bf584">

Is Svol the log(norm) of the target vector? If is, then I think maybe there's a mistake in computing the **log(norm) of the target vector**. This mistake might occur while secret vector and noise vector follows different distribution.

**Comment by lducas** [comment]
No, SVol is the (logarithm of) determinant of the ellipsoid that defines the error distribution.
